### ISSUE TITLE
Error Google Colab - Building Wheels

### ISSUE THREAD

**Issue by ClaudiaPPinheiro** [SimpleITK/SimpleITK#1627]
Hello! Since yesterday afternoon, when I try to install TorchIO on Google Colab, it always gets the same error while "building wheels for SimpleITK". It stays a long time trying to do it but always ends up throwing an error. During the day, I didn't have this problem, this was the first time and I've tried a lot of solutions to work it out without success.

**Comment by fepegar** [comment]
I have also noticed this in [TorchIO tests](https://github.com/fepegar/torchio/runs/5952172467?check_suite_focus=true).

I am going to forbid the new SimpleITK version in TorchIO requirements, so if you try again in a couple of hours it should work. Until then, you can run `%pip install "SimpleITK!=2.1.1.1"` before installing TorchIO on Colab.

**Comment by fepegar** [comment]
You shouldn't get the error on Colab anymore.

**Comment by ClaudiaPPinheiro** [comment]
Thank you! It worked :)

**Comment by jamesobutler** [comment]
@blowekamp Reviewing https://pypi.org/project/SimpleITK/2.1.1.1/#files, there are only Whl files for Python 3.6, 3.7 and 3.10. There are no Whl files for Python 3.8 and 3.9 which is why the GitHub workflow that I'm observing is taking a really long time now because it is trying to build the Whl from source.

**Comment by zivy** [comment]
Apologies to all.

During the upload of version 2.1.1.1, SimpleITK exceeded the allocated space on PyPI, after a partial upload. We have contacted the PyPI admins to exempt the project from the size limit constraint and are waiting to hear back.  

For now, when installing SimpleITK please explicitly specify an earlier version than 2.1.1.1 .

Will update here once we hear back from the PyPI folks.

**Comment by blowekamp** [comment]
I'd recommend using the following pip flag:
  [--prefer-binary ](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-prefer-binary)Prefer older binary packages over newer source packages.

The source distribution is available for 2.1.1.1, but not all compiled binary packages were able to be uploaded. pip is preferring the 2.1.1.1 source distribution over the precompiled 2.1.1 with out this flag.

**Comment by blowekamp** [comment]
The 2.1.1.1 source distribution has been deleted from Pypi.

**Comment by blowekamp** [comment]
The entire 2.1.1.1 release was yanked.

**Comment by Czaki** [comment]
> The entire 2.1.1.1 release was yanked.

Why? Why you do not leave cp 3.10 wheels? Why?

**Comment by blowekamp** [comment]
The release was yanking because several users reported the incomplete binary uploads were causing failures when installing that package

You can read more about the specification for yanking here:
https://blog.piwheels.org/new-features-deletion-yanking-and-more/

Yanking is different from deleting in that you can still download the binary when the specific version is specified. The following can temporarily be used in `requirements.txt` for 3.10 support:
```
SimpleITK==2.1.1.1; python_version == '3.10'
SimpleITK>=2.0
```

**Comment by jamesobutler** [comment]
@blowekamp Would it make sense to remove the Python 3.6 and 3.7 whls that were uploaded for 2.1.1.1 just to reduce confusion when reviewing the yanked release? The partial upload looks weird and it might make more sense if there were just Python 3.10 whls available which is why 2.1.1.1 was released in the first place.

**Comment by blowekamp** [comment]
Deleted the the non 3.10 wheels makes sense in the name of conserving used space.

Once a file is uploaded to PyPI it can not be changed. Once yanked or deleted it can not be restored or re uploaded. So the options are some what limited.

Also DIAGNijmegen/rse-panimg/#77 reported issues with the partial set of binaries with poetry.

Hopefully, The PPA quick increase the limit and SimpleITK 2.1.1.2 🥳 can be published with full binary support.

**Comment by jamesobutler** [comment]
@blowekamp With https://github.com/pypa/pypi-support/issues/1833 now merged, SimpleITK 2.1.1.2 can now be published?

**Comment by blowekamp** [comment]
SimpleITK 2.1.1.2 has just been pushed to PyPi.
